test/tests/pbs_admin_suspend.py

```diff
@@ -405,3 +413,416 @@ def test_degraded_resv(self):
+        self.server.expect(JOB, {'job_state': 'S'}, id=jid1)
+        self.server.expect(NODE, {'state': 'maintenance'}, id='vn[1]')
+
+        # submit other jobs asking for specific resources on vn[1]
+        j = Job(TEST_USER)
+        j.set_attributes({'foo': '2'})
+        jid2 = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid2)
+
+        # submit more jobs. They should be running
+        j = Job(TEST_USER)
+        jid3 = self.server.submit(j)
+        jid4 = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'R', 'substate': 42}, id=jid3)
+        self.server.expect(JOB, {'job_state': 'R', 'substate': 42}, id=jid4)
+
+        # verify that vn[1] is still in maintenance and
+        # job3 and job4 not running on vn[1]
+        self.server.expect(NODE, {'state': 'maintenance'}, id='vn[1]')
+        try:
+            self.server.expect(
+                JOB, {'exec_vnode': 'vn[1]'}, id=jid3, max_attempts=20)
+            self.server.expect(
+                JOB, {'exec_vnode': 'vn[1]'}, id=jid4, max_attempts=20)
+        except Exception, e:
+            self.assertFalse(e.rv)
+            print ("jid3 and jid4 not running on vn[1] as expected")
+
+    def test_list_jobs_1(self):
+        """
+        Test to list and set maintenance_jobs as various users
+        """
+
+        # submit a few jobs
+        j = Job(TEST_USER)
+        jid1 = self.server.submit(j)
+        jid2 = self.server.submit(j)
+        jid3 = self.server.submit(j)
+
+        # verify that all are running
+        self.server.expect(JOB, {'job_state=R': 3, 'substate=42': 3})
+
+        # admin-suspend 2 of them
+        self.server.sigjob(jid2, 'admin-suspend', runas=ROOT_USER)
+        self.server.sigjob(jid3, 'admin-suspend', runas=ROOT_USER)
+
+        # node state is in maintenance
+        self.server.expect(NODE, {'state': 'maintenance'}, id='vn[0]')
+
+        # list maintenance_jobs as root
+        self.server.expect(NODE, {'maintenance_jobs': jid2 + "," + jid3},
+                           runas=ROOT_USER)
+
+        # list maintenance jobs as user
+        self.server.expect(NODE, {'maintenance_jobs': jid2 + "," + jid3},
+                           runas=TEST_USER)
+
+        # set an operator
+        self.server.manager(MGR_CMD_SET, SERVER, {'operators': 'pbsoper@*'})
+
+        # List all jobs in maintenance mode as operator
+        self.server.expect(
+            NODE, {'maintenance_jobs': jid2 + "," + jid3}, runas='pbsoper')
+
+        # set maintenance_jobs as root
+        try:
+            self.server.manager(MGR_CMD_SET, NODE,
+                                {'maintenance_jobs': jid1}, id='vn[0]',
+                                runas=ROOT_USER)
+        except PbsManagerError, e:
+            self.assertFalse(e.rv)
+            msg = "Cannot set attribute, read only" +\
+                  " or insufficient permission  maintenance_jobs"
```

Comment:
I don't think this will work in API node.  I think qmgr is tagging on 'maintenance_jobs' to the end of that message.  If you care that this test passes in API mode, you might want to test it.